botocore/endpoint.py

```diff
@@ -294,9 +294,12 @@ def create_endpoint(self, service_model, region_name=None, is_secure=True,
                 raise
         # We only support the credentialScope.region in the properties
         # bag right now, so if it's available, it will override the
-        # provided region name.
-        region_name_override = endpoint['properties'].get(
-            'credentialScope', {}).get('region')
+        # provided region name if an endpoint url was not manually set.
+        # If a endpoint url was specified, the user must specify a region.
```

Comment:
I don't think that will fix the issue.  In the example above endpoint_url is not None so we don't pull in the region override from the heuristics file.